app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -190,7 +190,13 @@ public Void call() throws Exception {
                             } else {
                                 personDao.update(person);
                             }
-                        } catch (Exception e) {
+                            if (person.getEmail().equals(app.getEmail())) {
```

Comment:
@vishwesh3 I debugged and found that we are calling [setEmail](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/ZulipApp.java#L410) in [onRefresh](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java#L2109). Can you please check again and see if you are still getting this error?